src/Concerns/RedactsHeaders.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Laravel\Nightwatch\Concerns;
+
+use Symfony\Component\HttpFoundation\HeaderBag;
+use Throwable;
+
+use function array_map;
+use function explode;
+use function implode;
+use function in_array;
+use function strlen;
+use function strpos;
+use function strtolower;
+use function trim;
+
+trait RedactsHeaders
+{
+    /**
+     * @param  list<string>  $keys
+     */
+    private function redactHeaders(HeaderBag $headers, array $keys = []): HeaderBag
+    {
+        foreach ($keys as $key) {
+            if (! $headers->has($key)) {
+                continue;
+            }
+
+            $headers->set($key, array_map(fn ($value) => match (strtolower($key)) {
+                'authorization', 'proxy-authorization' => $this->redactAuthorizationHeaderValue((string) $value), // @phpstan-ignore cast.string
+                'cookie' => $this->redactCookieHeaderValue((string) $value), // @phpstan-ignore cast.string
+                default => $this->redactHeaderValue((string) $value), // @phpstan-ignore cast.string
+            }, $headers->all($key)));
+        }
+
+        return $headers;
+    }
+
+    private function redactHeaderValue(string $value): string
+    {
+        return '['.strlen($value).' bytes redacted]';
+    }
+
+    private function redactAuthorizationHeaderValue(string $value): string
+    {
+        if (strpos($value, ' ') === false) {
+            return $this->redactHeaderValue($value);
+        }
+
+        [$type, $remainder] = explode(' ', $value, 2);
```

Comment:
Does this need to handle when an authorization header does _not_ have a space in it?


```suggestion
        [$type, $remainder] = explode(' ', $value, 2) + [1 => null];
```